### PR TITLE
DashboardLinksSettings: Move them to Scenes

### LINKED ISSUE #80025: One can define and edit Dashboard Links
**The migration covers**:
* Show the list of links when going to dashboard settings > links
* Be able to edit them
* The dashboard detects the link changes and allows you to save them
* The compatible wrapper for `DashboardModel` supports `links`
* The serializer initializes the Scene when links are included in the Dashboard JSON
* The serializer serializes the same model when the scene contains links

In general, we try to reuse as many components as possible, so we will try to reuse the existing views

### PR DESCRIPTION
**Pull request by ivanortegaalba**
**The migration covers**:
* Show the list of links when going to dashboard settings > links
* Be able to edit them
* The dashboard detects the link changes and allows you to save them
* The compatible wrapper for `DashboardModel` supports `links`
* The serializer initializes the Scene when links are included in the Dashboard JSON
* The serializer serializes the same model when the scene contains links
* Refactor the previous components to be reusable. The no-scene link settings view is tested with RLT so they keep working as they were

Notes for the reviewer:
* I added rendering tests since the renderer contains logic that might break the experience

Fixes #80025

### CODE REVIEW

File: public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.test.ts [hunk 1]
```diff
@@ -817,15 +819,14 @@ describe('transformSceneToSaveModel', () => {
...
-      // TODO: Uncomment when we support links
-      // it('should remove links', async () => {
-      //   const scene = transformSaveModelToScene({ dashboard: snapshotableDashboardJson as any, meta: {} });
-      //   activateFullSceneTree(scene);
-      //   const snapshot = transformSceneToSaveModel(scene, true);
-      //   expect(snapshot.links?.length).toBe(1);
-      //   const result = trimDashboardForSnapshot('Snap title', getTimeRange({ from: 'now-6h', to: 'now' }), snapshot);
-      //   expect(result.links?.length).toBe(0);
-      // });
+      it('should remove links', async () => {
```
Comment (dprokop) [inline public/app/features/dashboard-scene/serialization/transformSceneToSaveModel.test.ts:822]: nice!

**Review by dprokop — APPROVED** [review]
Amazing work @ivanortegaalba! Love the tests, very well done.

I've only noticed one issue when navigating back from individual link edit to the links list - instead of going back to the list, the navigation ends up on the General settings.

File: public/app/features/dashboard/components/LinksSettings/DashboardLinkForm.tsx [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+const linkIconOptions = Object.keys(linkIconMap).map((key) => ({ label: key, value: key }));
+
+interface DashboardLinkFormProps {
+  link: DashboardLink;
+  onUpdate: (link: DashboardLink) => void;
+  onGoBack: () => void;
+}
+
+export function DashboardLinkForm({ link, onUpdate, onGoBack }: DashboardLinkFormProps) {
```
[the remark this answers is not in the document]
Comment (ivanortegaalba) [reply public/app/features/dashboard/components/LinksSettings/DashboardLinkForm.tsx:22]: Sure I can! I will only move the form and the list, so they won't be removed. Thanks for the catch!

File: public/app/features/dashboard/components/LinksSettings/DashboardLinkList.tsx [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+              <td style={{ width: '1%' }} role="gridcell">
+                {links.length > 1 && idx !== links.length - 1 ? (
+                  <IconButton name="arrow-down" onClick={() => onOrderChange(idx, 1)} tooltip="Move link down" />
+                ) : null}
+              </td>
+              <td style={{ width: '1%' }} role="gridcell">
+                <IconButton name="copy" onClick={() => onDuplicate(link)} tooltip="Copy link" />
+              </td>
+              <td style={{ width: '1%' }} role="gridcell">
+                <DeleteButton
```
Comment (dprokop) [inline public/app/features/dashboard/components/LinksSettings/DashboardLinkList.tsx:85]: @ivanortegaalba - one minor thing me and @axelavargas just thought about. Maybe we could unify this deletion flow with the one that we have for variables? Basically a thrash icon with a confirmation modal. That would be a nice unification across. the settings views.

**Pull request merged**